### PR TITLE
Add Tensorflow SavedBundleModel support.

### PR DESCRIPTION
**Pull request by regadas**
Still misses SavedBundleModel tests ... Want to gather some feedback.

### CODE REVIEW

File: scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowSCollectionFunctions.scala [hunk 1]
```diff
@@ -17,167 +17,208 @@
...
+    * @param options   configuration parameters for the session specified as a
+    *                 `com.spotify.zoltar.tf.TensorFlowModel.Options`.
+    * @param inFn     translates input elements of T to map of input-operation ->
+    *                 [[org.tensorflow.Tensor Tensor]]. This method takes ownership of the
+    *                 [[org.tensorflow.Tensor Tensor]]s.
+    * @param outFn    translates output of prediction from map of output-operation ->
+    *                 [[org.tensorflow.Tensor Tensor]], to elements of V. This method takes
+    *                 ownership of the [[org.tensorflow.Tensor Tensor]]s.
+    */
+  def predict[V: ClassTag, W](uri: String,
```
Comment (regadas) [inline scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowSCollectionFunctions.scala]: @ravwojdyla not sure about the naming here. Can it be a bit confusing? 

probably we could type this `uri` like ... `SavedBundleModelUri(uri)` wdyt?

File: scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowSCollectionFunctions.scala [hunk 2]
```diff
@@ -17,167 +17,208 @@
...
    */
-  def predict[V: ClassTag, W](graphUri: String,
+  def predict[V: ClassTag, W](graphUri: String, fetchOps: Seq[String], config: Array[Byte] = null)(
+    inFn: T => Map[String, Tensor[_]])(outFn: (T, Map[String, Tensor[_]]) => V): SCollection[V] =
+    self.parDo(new GraphPredictDoFn[T, V](graphUri, fetchOps, config, inFn, outFn))
+
+  /**
+    * Predict/infer/forward-pass on pre-trained SaveModelBundle.
+    *
+    * @param uri URI of pre-trained/saved TensorFlow model
```
Comment (ravwojdyla) [inline scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowSCollectionFunctions.scala]: TensorFlow `Saved Model` directory

File: scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowSCollectionFunctions.scala [hunk 3]
```diff
@@ -17,167 +17,208 @@
...
+   *                 [[org.tensorflow.Tensor Tensor]], to elements of V. This method takes
+   *                 ownership of the [[org.tensorflow.Tensor Tensor]]s.
    */
-  def predict[V: ClassTag, W](graphUri: String,
+  def predict[V: ClassTag, W](graphUri: String, fetchOps: Seq[String], config: Array[Byte] = null)(
+    inFn: T => Map[String, Tensor[_]])(outFn: (T, Map[String, Tensor[_]]) => V): SCollection[V] =
+    self.parDo(new GraphPredictDoFn[T, V](graphUri, fetchOps, config, inFn, outFn))
+
+  /**
+    * Predict/infer/forward-pass on pre-trained SaveModelBundle.
```
Comment (ravwojdyla) [inline scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowSCollectionFunctions.scala]: `SaveModelBundle` -> `Saved Model`.

File: scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowSCollectionFunctions.scala (hunk 1 above)
Comment (ravwojdyla) [reply scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowSCollectionFunctions.scala]: `uri` -> `dirPath` or `path`

File: build.sbt [hunk 4]
```diff
@@ -65,6 +65,7 @@ val shapelessDatatypeVersion = "0.1.9"
 val slf4jVersion = "1.7.25"
 val sparkeyVersion = "2.3.0"
 val tensorFlowVersion = "1.8.0"
+val zoltarVersion = "0.3.1-SNAPSHOT"
```
Comment (regadas) [inline build.sbt]: I think this just needs to wait for the next `zoltar` release.

File: scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowFunctions.scala [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+   */
+  @deprecated("Tensorflow Graph model support will be removed. Use Saved Model predict.",
+              "scio-tensorflow 0.5.4")
+  def predict[V: ClassTag, W](graphUri: String, fetchOps: Seq[String], config: Array[Byte] = null)(
+    inFn: T => Map[String, Tensor[_]])(outFn: (T, Map[String, Tensor[_]]) => V): SCollection[V] =
+    self.parDo(new GraphPredictDoFn[T, V](graphUri, fetchOps, config, inFn, outFn))
+
+}
+
+class TFExampleSCollectionFunctions[T <: Example](val self: SCollection[T]) {
```
Comment (regadas) [inline scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowFunctions.scala]: @ravwojdyla I think this classes can be `private[tensorflow]`. wdyt?

File: scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowImplicits.scala [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+   * Implicit conversion from [[com.spotify.scio.values.SCollection SCollection]] to
+   * [[TFExampleSCollectionFunctions]].
+   */
+  implicit def makeSeqTFExampleSCollectionFunctions[T <: Example](
+    s: SCollection[Seq[T]]): SeqTFExampleSCollectionFunctions[T] =
+    new SeqTFExampleSCollectionFunctions(s)
+
+}
+
+object TensorFlowImplicits extends TensorFlowImplicits
```
Comment (ravwojdyla) [inline scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowImplicits.scala]: Could you pls comment this trick you are doing here with `trait` and `package object` - for future generations.

File: scio-tensorflow/src/test/scala/com/spotify/scio/tensorflow/TensorflowSpec.scala [hunk 7]
```diff
@@ -18,33 +18,90 @@
...
+    val collection =
+      sc.parallelize(List(Iris(Some(5.1), Some(3.5), Some(1.4), Some(0.2), Some("Iris-setosa"))))
+
+    Spec
+      .extractWithSettings(collection, settings)
+      .featureValues[Example]
+      .predict(args("savedModelUri"), Seq("linear/head/predictions/class_ids"), options) { e =>
+        Map("input_example_tensor" -> Tensors.create(Array(e.toByteArray)))
+      } { (r, o) =>
+        (r, o.map {
```
Comment (ravwojdyla) [inline scio-tensorflow/src/test/scala/com/spotify/scio/tensorflow/TensorflowSpec.scala]: We probably could use `JTensor` here, but now looking at `JTensor` it doesn't seem like it's `Serializable` :/
https://github.com/spotify/zoltar/issues/104

Comment (regadas) [reply scio-tensorflow/src/test/scala/com/spotify/scio/tensorflow/TensorflowSpec.scala]: @ravwojdyla yeah that would be nicer i guess!

File: scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowFunctions.scala [hunk 8]
```diff
@@ -0,0 +1,342 @@
...
+private[tensorflow] class GraphPredictDoFn[T, V](uri: String,
+                                                 fetchOp: Seq[String],
+                                                 @Nullable config: Array[Byte],
+                                                 inFn: T => Map[String, Tensor[_]],
+                                                 outFn: (T, Map[String, Tensor[_]]) => V)
+    extends PredictDoFn[T, V, TensorFlowGraphModel](fetchOp, inFn, outFn) {
+  @transient private lazy val log = LoggerFactory.getLogger(this.getClass)
+
+  override def createResource(): ModelLoader[TensorFlowGraphModel] = {
+    log.info("Loading TensorFlow graph")
```
[the remark this answers is not in the document]
Comment (regadas) [reply scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowFunctions.scala]: probably not

**Comment by regadas** [comment]
Closing in favour of #1162

**Pull request closed without merging**